src/jit/codegenarmarch.cpp

```diff
@@ -727,8 +743,9 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
             gcPtrCount = treeNode->gtNumberReferenceSlots;
 #endif
             // Setup the structSize, isHFa, and gcPtrCount
-            if (varNode != nullptr)
+            if (source->OperGet() == GT_LCL_VAR)
```

Comment:
I don't think this will work because `varNumInp` is no longer set when you have `OBJ(LCL_VAR_ADDR)`.

And the `GT_LCL_VAR` case doesn't happen anyway. Well, I'm 99.9% sure about that.

Reply:
It might be good to just get rid of `varNumInp` as the only user can be changed to use `varNode->GetLclNum()`.

Reply:
Lets try.

Reply:
> It might be good to just get rid of varNumInp as the only user can be changed to use varNode->GetLclNum().

It works, the alternative would be to explicitly disable the warning for the affected lines or add a fake init ( `#define DUMMY_INIT(x) (x)`).

Reply:
Looks pretty good as it is, I don't think it's necessary to go the fake init way.